repository/github_user.go

```diff
@@ -0,0 +1,96 @@
+//Package repository container the fuctions to handle githubusers model information
+package repository
+
+import (
+	"encoding/csv"
+	"io"
+	"log"
+	"lolidelgado/github-users/models"
+	"os"
+	"strconv"
+)
+
+type GithubUser struct {
+	githubUserCsvFileName string
+}
+
+type IGithubUserRepository interface {
+	FetchAll() ([]models.GithubUser, error)
+}
+
+const staticFilesPath = "static/"
+
+func NewGithubUser(fileName string) *GithubUser {
+	return &GithubUser{
+		fileName,
+	}
+}
+
+func (g *GithubUser) FetchAll() ([]models.GithubUser, error) {
+	lines, err := g.readCsv()
+	if err != nil {
+		return nil, err
+	}
+	return arrayToGithubUser(lines), nil
+}
+
+func (g *GithubUser) GetById(id int) ([]models.GithubUser, error) {
+	lines, err := g.readCsv()
+	if err != nil {
+		return nil, err
+	}
+	var user [][]string
+	for _, line := range lines {
+		if idFromFile, _ := strconv.Atoi(line[0]); idFromFile == id {
+			user = append(user, line)
+			break
+		}
+	}
+	if len(user) > 0 {
+		return arrayToGithubUser(user), nil
+	} else {
+		return []models.GithubUser{}, nil
+	}
+}
+
+func (g *GithubUser) readCsv() ([][]string, error) {
+	//csv reader
+	file, err := os.Open(staticFilesPath + g.githubUserCsvFileName)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+	csvReader := csv.NewReader(file)
+
+	var lines [][]string
+	for {
+		record, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.Fatal(err)
```

Comment:
You should not call log.Fatal here.

Normally you back-propagate the error to the caller of this function; in this case it was in the same package, but also that function was called on the usecase, which was called by the controller.

Also, if you call `log.Fatal` here, the defer in line 62 will never be called.

Controller should be the one that handles this error; and avoiding ending the service execution abruptly.